sw/device/lib/testing/test_framework/ujson_ottf.h

```diff
@@ -26,12 +27,14 @@ ujson_t ujson_ottf_console(void);
  * @param uj_ctx_ A `ujson_t` representing the IO context.
  * @param data_ A pointer to the data to send.
  */
-#define RESP_OK(responder_, uj_ctx_, data_)    \
-  ({                                           \
-    TRY(ujson_putbuf(uj_ctx_, "RESP_OK:", 8)); \
-    TRY(responder_(uj_ctx_, data_));           \
-    TRY(ujson_putbuf(uj_ctx_, "\r\n", 2));     \
-    OK_STATUS();                               \
+#define RESP_OK(responder_, uj_ctx_, data_)                      \
+  ({                                                             \
+    TRY(ujson_putbuf(uj_ctx_, "RESP_OK:", 8));                   \
+    ujson_crc32_reset(uj_ctx_);                                  \
+    TRY(responder_(uj_ctx_, data_));                             \
+    base_fprintf(ujson_into_buffer_sink(uj_ctx_), "\nCRC: %x\n", \
```

Comment:
What if you just had a function `ujson_crc32_send()` that did the the `base_fprintf` call and then you wouldn't need to expose the function `ujson_into_buffer_sink()`? I think the point of the ujson framework is to keep things JSON formatted and prevent exposing an easy way to grab a buffer sink that you can printf too (even though that is what you need to do here essentially hehe).

Reply:
Currently, the ujson protocol is line-based.  I'd advocate for using a space character rather than a new-line to separate the end of the JSON message from the CRC.

To be nit-picky, I'd also print the CRC as "CRC:%x", omitting the space between the CRC identifier and the value.

Lastly, if you wanted to avoid `printf` altogether here, you can use the existing ujson functions (although, this changes the representation from hex to decimal):
```
uint32_t crc = ujson_crc32_finish(uj_ctx_);
TRY(ujson_putbuf(uj_ctx_, " CRC:", 4));
TRY(ujson_serialize_uint32_t(uj_ctx_, crc));
TRY(ujson_putbuf(uj_ctx_, "\n", 1));
```
(Honestly, I'm not sure if avoiding `printf` is really that important.  We're already using because it can convert `status_t`).

I'd probably put that in a helper function so you can just call it in the various `RESP` macros.

Reply:
I decided to forgo the pretty hex and just use `ujson_serialize_uint32_t`. Also, changed CRC format to Chris' preference.

I think it would be neat to remove the `printf` dependency and use the implementation suggested [here](https://github.com/lowRISC/opentitan/pull/18914#issue-1757164757), but probably not worth the time.